epochStart/interface.go

```diff
@@ -129,7 +129,7 @@ type StartOfEpochMetaSyncer interface {
 
 // NodesConfigProvider will provide the necessary information for start in epoch economics block creation
 type NodesConfigProvider interface {
-	ConsensusGroupSize(shardID uint32) int
+	ConsensusGroupSize(shardID uint32, epoch uint32) int
```

Comment:
I believe this is a bit dangerous as I can call the function as
```GO
NodesConfigProvider.ConsensusGroupSize(epoch, shardID)
```
and the compiler will be just happy. Other than using a special argument DTO struct which will complicate things, the only option is to have the epoch as something else than a uint32. But that will need a lot of useless casts.

Reply:
renamed to `ConsensusGroupSizeForShardAndEpoch`